remapping_pipeline.nf

```diff
@@ -0,0 +1,350 @@
+#!/usr/bin/env nextflow
+
+def helpMessage() {
+    log.info"""
+    Index the provided genome with bowtie and samtools
+    Usage:
+            --oldgenome                     Fasta file containing the reference genome.
+            --outdir                    Directory where the index will be deployed.
+    """
+}
+params.flankingseq = 50
+params.scorecutoff = 0.6
+params.diffcutoff =  0.04
+params.outfile = "$baseDir/resources/remap_vcf.vcf"
+
+// Show help message
+if (params.help) exit 0, helpMessage()
+params.oldgenome = "$baseDir/resources/genome.fa"
+params.newgenome = "$baseDir/resources/genome.fa"
+params.vcffile = "$baseDir/resources/source_vcf.vcf"
+
+// Index files for both old and new genomes 
+oldgenome_chrom_sizes = file("${params.oldgenome}.chrom.sizes")
+oldgenome_fai = file("${params.oldgenome}.fai")
+newgenome_dir = file(params.newgenome).getParent()
+newgenome_basename = file(params.newgenome).getName()
+newgenome_fai = file("${params.newgenome}.fai")
+
+// basename and basedir of the output file to know how to name the output file
+outfile_basename = file(params.outfile).getName()
+outfile_dir = file(params.outfile).getParent()
+
+
+/*
+ * Store the original VCF header for later use
+ */
+process StoreVCFHeader {
+
+    input:
+        path "source_vcf" from params.vcffile
+
+    output:
+        path "vcf_header.txt" into vcf_header
+
+    """
+    bcftools view --header-only source_vcf -o vcf_header.txt
+    """
+}
+
+/*
+ * Convert vcf file to bed format
+ */
+process ConvertVCFToBed {
+
+    input:
+        path "source_vcf" from params.vcffile
+
+    output:
+        path "variants.bed" into variants_bed
+
+    """
+    vcf2bed < source_vcf > variants.bed
+    """
+}
```

Comment:
@tcezard @BaronKoy point for discussion — I wonder if we should aim for more compact representation, at least for simple commands? Compare:

```nextflow
process ConvertVCFToBed {

    input:
        path "source_vcf" from params.vcffile

    output:
        path "variants.bed" into variants_bed

    """
    vcf2bed < source_vcf > variants.bed
    """
}
```

And:

```nextflow
process ConvertVCFToBed {
    input: path "source_vcf" from params.vcffile
    output: path "variants.bed" into variants_bed
    "vcf2bed < source_vcf > variants.bed"
}
```

The first one is 12 lines and the second one is 5. Also, the second one feels more easily readable to me. What do you think?

Reply:
I actually prefer if with all the extra new lines, but that clearly a matter of opinion.
More importantly I think we should be consistent across processes and the more airy version is more applicable to the general case than the compact one.

Reply:
Actually, on the second look, I now also prefer the first, less condensed option. Don't know what got into me yesterday :) Let's use the first approach consistently for everything.